### ISSUE TITLE
Update image and redeploy service

### ISSUE THREAD

**Issue by bee-keeper** [aws/amazon-ecs-cli#320]
Hi there,

I'm a bit confused about the direction of this project and purpose of this tool.

I've been looking a 3rd party solutions to the simple problem of updating a task with a newly tagged image and then redeploying the service.  AWS command line does this fairly nicely and we can use the JSON definitions we're accustomed to seeing in the web interface.  There are also a number of third party projects which aim to fill this hole, for instance ecs-deploy which allows you to do something like `ecs deploy my-cluster my-service --tag 1.2.3`.

So I was somewhat surprised with ecs-cli there doesn't seem to be any mention of using it for deploys and furthermore it introduces the idea of using compose files for configuration rather than JSON which seems to be how everything is configured on the web interface.  At this moment in time I can't see any advantage to using it over aws-cli for deployments.  It seems much more cumbersome.

So in short, my questions are, why Compose over JSON?  Should I carry on with using aws-cli for instructing ECS to do deploys or is the suggested AWS preference to use ecs-cli?  And if we can easily use ecs-cli for deployments please can we get some examples in the docs as this is pretty much the primary objective for anyone using ECS.

**Comment by SoManyHs** [comment]
Hi @bee-keeper,
 
When ECS was originally designed, we wanted to make our customer experience as streamlined as possible with existing tooling in the container space while avoiding redundancy with the AWS CLI. Since Docker was and continues to be the most widespread container technology, most adopters of a containerized ecosystem were mostly familiar with docker-compose as their orchestration engine. We wanted to provide value for those customers by allowing them to switch over to ECS without the extra overhead of learning a new container definition syntax (e.g. ECS Task Definition JSON).
 
The goal of the ECS CLI is to simplify workflows that use ECS. One example is our [cluster up]( http://docs.aws.amazon.com/AmazonECS/latest/developerguide/cmd-ecs-cli-up.html) command, which creates an ECS cluster and all the AWS resources needed to set it up (including EC2 instances, the latest ECS Agent, IAM Roles, etc) in a single step; or [compose up]( http://docs.aws.amazon.com/AmazonECS/latest/developerguide/cmd-ecs-cli-compose.html), which creates a task definition from a docker-compose file and starts a task on your cluster. Of course, we are still working to incorporate more workflows into the ECS CLI, so you may find the AWS CLI to be more suited to your needs in the meantime, especially if you're primarily using Task Definition JSON files rather than Compose files.
 
The deployment use case you are describing is currently provided through [compose service up]( http://docs.aws.amazon.com/AmazonECS/latest/developerguide/cmd-ecs-cli-compose-service.html), though as you pointed out this is still dependent on using the docker-compose file format. 

As the container ecosystem evolves, we will continue to adapt our CLI to best suit the needs of customers. Currently, we are working on a roadmap for the CLI that has a user experience that is decoupled from docker-compose and emphasizing the use of Task Definitions as a primitive.
 
Thanks for bringing up your concerns. Let us know if there’s anything else we can answer for you.

**Comment by bee-keeper** [comment]
Hi, thanks for the extensive reply.  

It feels more a general issue for AWS that firstly ECS tooling is still quite patchy (ie no single tool is anywhere near feature complete) and secondly the docs don't seem to have anything regarding deployments at all.  Coming from Kubernetes I've been quite surprised that this is the case.

**Comment by yinshiua** [comment]
@bee-keeper We understand your frustration and we are actively working on improving the developer experience, supporting a full set of ECS features, and supporting Task Definition as a primitive in the ECS CLI. Please let us know if you have any constructive feedback that you would like us to incorporate into ECS CLI.
